pkg/cmd/update-tls-artifacts/generate-owners/tlsmetadata/autoregenerate_after_expiry/requirement.go

```diff
@@ -10,68 +9,20 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-// TODO move to openshift/api
-const AutoRegenerateAfterOfflineExpiryAnnotation = "certificates.openshift.io/auto-regenerate-after-offline-expiry"
+const annotationName string = "certificates.openshift.io/auto-regenerate-after-offline-expiry"
 
-type AutoRegenerateAfterOfflineExpiryRequirement struct {
-	name string
-}
+// TODO move to openshift/api
+type AutoRegenerateAfterOfflineExpiryRequirement struct{}
 
 func NewAutoRegenerateAfterOfflineExpiryRequirement() tlsmetadatainterfaces.Requirement {
-	return AutoRegenerateAfterOfflineExpiryRequirement{
-		name: "autoregenerate-after-expiry",
-	}
-}
-
-func (o AutoRegenerateAfterOfflineExpiryRequirement) InspectRequirement(rawData []*certgraphapi.PKIList) (tlsmetadatainterfaces.RequirementResult, error) {
-	pkiInfo, err := tlsmetadatainterfaces.ProcessByLocation(rawData)
-	if err != nil {
-		return nil, fmt.Errorf("transforming raw data %v: %w", o.GetName(), err)
-	}
-
-	ownershipJSONBytes, err := json.MarshalIndent(pkiInfo, "", "    ")
-	if err != nil {
-		return nil, fmt.Errorf("failure marshalling %v.json: %w", o.GetName(), err)
-	}
-	markdown, err := generateAutoRegenerateAfterOfflineExpiryshipMarkdown(pkiInfo)
-	if err != nil {
-		return nil, fmt.Errorf("failure marshalling %v.md: %w", o.GetName(), err)
-	}
-	violations := generateViolationJSON(pkiInfo)
-	violationJSONBytes, err := json.MarshalIndent(violations, "", "    ")
-	if err != nil {
-		return nil, fmt.Errorf("failure marshalling %v-violations.json: %w", o.GetName(), err)
-	}
-
-	return tlsmetadatainterfaces.NewRequirementResult(
-		o.GetName(),
-		ownershipJSONBytes,
-		markdown,
-		violationJSONBytes)
-}
-
-func generateViolationJSON(pkiInfo *certgraphapi.PKIRegistryInfo) *certgraphapi.PKIRegistryInfo {
-	ret := &certgraphapi.PKIRegistryInfo{}
-
-	for i := range pkiInfo.CertKeyPairs {
-		curr := pkiInfo.CertKeyPairs[i]
-		regenerates, _ := tlsmetadatainterfaces.AnnotationValue(curr.CertKeyInfo.SelectedCertMetadataAnnotations, AutoRegenerateAfterOfflineExpiryAnnotation)
-		if len(regenerates) == 0 {
-			ret.CertKeyPairs = append(ret.CertKeyPairs, curr)
-		}
-	}
-	for i := range pkiInfo.CertificateAuthorityBundles {
-		curr := pkiInfo.CertificateAuthorityBundles[i]
-		regenerates, _ := tlsmetadatainterfaces.AnnotationValue(curr.CABundleInfo.SelectedCertMetadataAnnotations, AutoRegenerateAfterOfflineExpiryAnnotation)
-		if len(regenerates) == 0 {
-			ret.CertificateAuthorityBundles = append(ret.CertificateAuthorityBundles, curr)
-		}
-	}
-
-	return ret
+	return tlsmetadatainterfaces.NewAnnotationRequirement(
+		"autoregenerate-after-expiry",
+		annotationName,
+		generateAutoRegenerateAfterOfflineExpiryMarkdownFn,
```

Comment:
Probably just need the "what does this mean" block, right?

Reply:
Updated, PTAL